pkg/machine/ocipull/ociartifact.go

```diff
@@ -114,36 +118,44 @@ func (o *OCIArtifactDisk) OriginalFileName() (string, string) {
 }
 
 func (o *OCIArtifactDisk) Get() error {
-	if err := o.get(); err != nil {
+	cleanCache, err := o.get()
+	if err != nil {
 		return err
 	}
+	defer cleanCache()
```

Comment:
you never check cleanCache != nil, but in your error paths the function may be very well return nil instead of a function causing a panic